### DISCUSSION TITLE
404 Not found

### DISCUSSION THREAD

**Discussion by jamesmiddendorff** [InnovateAsterisk/Browser-Phone#472] [General]
Hello!

I have pretty much followed your tutorial and when I try to access my server from a browser I see:
Not Found
The requested URL was not found on this server.

Asterisk/16.28.0~dfsg-0+deb11u3

One thing I have not done was to setup the certificates.  I am using this for my http.conf:

                                                      /etc/asterisk/http.conf                                                                     
[general]
enabled=yes
bindaddr=0.0.0.0
bindport=8088
tlsenable=no
enablestatic=yes
redirect=/ /static/index.html

Do you have to use the certificates? I am trying to build a system that can be deployed via an image to our support technicians so they can lab a pbx at home with our specific system.

thanks for any tips and great work!

**Comment by InnovateAsterisk** [comment]
Well, it shouldn't say "Not Found". If you are not using the certificates the phone will display, but not be able to connect to the Server.

I'm assuming you are loading the url as ```http://server_name:8088/```, correct?

> Do you have to use the certificates?

Well the thing is this; Most major browsers don't allow you to access the microphone without the page being hosted over an SSL connection. Then the same for the Websocket connection, it can connect without an SSL, but the getUserMedia() will fail (the call to capture the microphone). So even if the index.html page is loaded over HTTP, the code will not work.

Then it gets more tricky (if you are not using localhost), the certificate you make can't be a self-signed certificate (as of a few years ago). You actually have to make a self-signed (made up) Certificate Authority. This is different to a  _self-signed certificate_, its the actual certificate used to sign client certificates. It's that CA certificate that you actually need to deploy (install) on each client PC, that will make that client certificate valid.

There are workarounds for some things. Firstly, there is this: you don't actually have to host the html & javascripts on your actual Asterisk box (unless its not connected to the internet at all), you simply need to loaded this page https://www.innovateasterisk.com/phone/ from any location, and you can put in the settings of your lab PBX, and it will connect... so long as (and here is the tricky part again). When you specify server location (of your lab PBX) ```Secure WebSocket Server (TLS):``` this now again has to be served with a proper certificate. This means it cant be an IP address, and it again, can't be self-signed.

If you do generate a self-signed client certificate for the HTTP page, I seem to remember there was a way to "accept the risk" but it prompts a pretty big warning, and most people would shy away from the warning screen. I seem to remember some browsers working differently around this too, so this isn't the advised route.
